app/screens/convert_gm_to_channel/convert_gm_to_channel_form/convert_gm_to_channel_form.tsx

```diff
@@ -0,0 +1,137 @@
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useState} from 'react';
+import {useIntl} from 'react-intl';
+import {View} from 'react-native';
+
+import {convertGroupMessageToPrivateChannel, switchToChannelById} from '@actions/remote/channel';
+import {useServerUrl} from '@app/context/server';
+import {useTheme} from '@app/context/theme';
+import {logError} from '@app/utils/log';
+import {makeStyleSheetFromTheme} from '@app/utils/theme';
+import {displayUsername} from '@app/utils/user';
+import Button from '@components/button';
+
+import {ChannelNameInput} from '../channel_name_input';
+import MessageBox from '../message_box/message_box';
+import {TeamSelector} from '../team_selector';
+
+import {NoCommonTeamForm} from './no_common_teams_form';
+
+const getStyleFromTheme = makeStyleSheetFromTheme(() => {
+    return {
+        container: {
+            paddingVertical: 24,
+            paddingHorizontal: 20,
+            display: 'flex',
+            flexDirection: 'column',
+            gap: 24,
+        },
+    };
+});
+
+type Props = {
+    channelId: string;
+    commonTeams: Team[];
+    profiles: UserProfile[];
+    locale?: string;
+    teammateNameDisplay?: string;
+}
+
+export const ConvertGMToChannelForm = ({
+    channelId,
+    commonTeams,
+    profiles,
+    locale,
+    teammateNameDisplay,
+}: Props) => {
+    const theme = useTheme();
+    const styles = getStyleFromTheme(theme);
+    const serverUrl = useServerUrl();
+    const intl = useIntl();
+
+    const [selectedTeam, setSelectedTeam] = useState<Team>();
+    const [newChannelName, setNewChannelName] = useState<string>('');
+
+    const submitButtonEnabled = selectedTeam && newChannelName.trim();
+
+    const {formatMessage} = useIntl();
+    const confirmButtonText = formatMessage({
+        id: 'channel_info.convert_gm_to_channel.button_text',
+        defaultMessage: 'Convert to Private Channel',
+    });
+
+    useEffect(() => {
+        if (commonTeams.length > 0) {
+            setSelectedTeam(commonTeams[0]);
+        }
+    }, [commonTeams]);
+
+    const handleOnPress = useCallback(async () => {
+        if (!submitButtonEnabled) {
+            return;
+        }
+
+        const {updatedChannel, error} = await convertGroupMessageToPrivateChannel(serverUrl, channelId, selectedTeam.id, newChannelName);
+
+        if (error) {
+            logError(error);
+            return;
+        }
+
+        if (!updatedChannel) {
+            return;
+        }
+
+        await switchToChannelById(serverUrl, updatedChannel.id, selectedTeam.id);
+    }, [selectedTeam, newChannelName, submitButtonEnabled]);
+
+    const messageBoxHeader = intl.formatMessage({
+        id: 'channel_info.convert_gm_to_channel.warning.header',
+        defaultMessage: 'Conversation history will be visible to any channel members',
+    });
+
+    const userDisplayNames = profiles.map((profile) => displayUsername(profile, locale, teammateNameDisplay));
```

Comment:
this could use a `useMemo` to avoid iteration every time the component re-renders.

Reply:
If we do that, we might display stale usernames if someone updates it when its open. Since its only 7 users max, I think its fine. Thoughts? @larkox we had the same discussion in context on webapp

Reply:
Thanks for the explanation @larkox  . Added useMemo.